app/src/main/java/org/sopt/and/sign/signin/SignInScreen.kt

```diff
@@ -58,22 +58,20 @@ fun SignInScreen(
     viewModel: SignInViewModel = viewModel()
 ){
 
-    val state by viewModel.state.collectAsState()
-
-    val context = LocalContext.current
-    val preferenceUtil = PreferenceUtil(context)
+    val state by viewModel.state.collectAsStateWithLifecycle()
+    val context = rememberUpdatedState(LocalContext.current).value
     val snackBarHostState = remember { SnackbarHostState() }
 
     LaunchedEffect(viewModel.intent) {
        viewModel.intent.collect{ intent ->
```

Comment:
SharedFlow도 lifecycle에 맞춰서 안전하게 수집할 수 있습니다!
```suggestion
       viewModel.intent.flowWithLifecycle(LocalLifecycleOwner.lifecycle)
             .collect{ intent ->
```

Reply:
4주차에 반영해보겠습니다😄